contracts/PBAB+Collabs/GenArt721CoreV2WithEvents.sol

```diff
@@ -0,0 +1,758 @@
+     */
+    function updateProjectSecondaryMarketRoyaltyPercentage(
+        uint256 _projectId,
+        uint256 _secondMarketRoyalty
+    ) public onlyArtist(_projectId) {
+        require(_secondMarketRoyalty <= 100, "Max of 100%");
+        projectIdToSecondaryMarketRoyaltyPercentage[
+            _projectId
+        ] = _secondMarketRoyalty;
+
+        emit ProjectUpdated(_projectId, "secondaryMarketRoyaltyPercentage");
+    }
+
+    /**
+     * @notice Updates description of project `_projectId`.
+     */
+    function updateProjectDescription(
+        uint256 _projectId,
+        string memory _projectDescription
+    ) public onlyArtist(_projectId) {
+        projects[_projectId].description = _projectDescription;
+
+        emit ProjectUpdated(_projectId, "projectDescription");
+    }
+
+    /**
+     * @notice Updates website of project `_projectId` to be `_projectWebsite`.
+     */
+    function updateProjectWebsite(
+        uint256 _projectId,
+        string memory _projectWebsite
+    ) public onlyArtist(_projectId) {
+        projects[_projectId].website = _projectWebsite;
+
+        emit ProjectUpdated(_projectId, "projectWebsite");
+    }
+
+    /**
+     * @notice Updates license for project `_projectId`.
+     */
+    function updateProjectLicense(
+        uint256 _projectId,
+        string memory _projectLicense
+    ) public onlyUnlocked(_projectId) onlyArtistOrWhitelisted(_projectId) {
+        projects[_projectId].license = _projectLicense;
+
+        emit ProjectUpdated(_projectId, "projectLicense");
+    }
+
+    /**
+     * @notice Updates maximum invocations for project `_projectId` to
+     * `_maxInvocations`.
+     */
+    function updateProjectMaxInvocations(
+        uint256 _projectId,
+        uint256 _maxInvocations
+    ) public onlyArtist(_projectId) {
+        require(
+            (!projects[_projectId].locked ||
+                _maxInvocations < projects[_projectId].maxInvocations),
+            "Only if unlocked"
+        );
+        require(
+            _maxInvocations > projects[_projectId].invocations,
+            "You must set max invocations greater than current invocations"
+        );
+        require(_maxInvocations <= ONE_MILLION, "Cannot exceed 1000000");
+        projects[_projectId].maxInvocations = _maxInvocations;
+
+        emit ProjectUpdated(_projectId, "projectMaxInvocations");
+    }
+
+    /**
+     * @notice Adds a script to project `_projectId`.
+     * @param _projectId Project to be updated.
+     * @param _script Script to be added.
+     */
+    function addProjectScript(uint256 _projectId, string memory _script)
+        public
+        onlyUnlocked(_projectId)
+        onlyArtistOrWhitelisted(_projectId)
+    {
+        projects[_projectId].scripts[
+            projects[_projectId].scriptCount
+        ] = _script;
+        projects[_projectId].scriptCount = projects[_projectId].scriptCount + 1;
+
+        emit ProjectUpdated(_projectId, "addProjectScript");
```

Comment:
Since we always refresh the whole script in the subgraph let's just make the field here "script"